src/app/app.component.ts

```diff
@@ -156,10 +156,19 @@ export class AppComponent {
   }
 
   /**
-   * Run app-specific first launch tasks
+   * Run app-specific launch tasks
    **/
-  private async handleFirstLaunchDataActions() {
-    for (const initAction of APP_INITIALISATION_DEFAULTS.app_first_launch_actions) {
+  private async handleLaunchDataActions(user) {
+    for (const initAction of APP_INITIALISATION_DEFAULTS.app_launch_actions) {
+      if (initAction.first_launch_only && user.first_app_open) {
+        continue;
+      }
+      if (initAction.condition) {
+        const conditionMet = this.checkLaunchActionCondition(initAction.condition, user);
```

Comment:
There is `evaluateConditionString` method in the template service that can be used to parse any condition (containing dynamic text or not). This would probably be better than define another condition handler, and enable using `@fields.some_field="some_value"`.